File: java/media/Controller.java
import java.io.OutputStream;
import java.net.Socket;

public class Controller {
    public static void main(String[] args) throws Exception {
        try (Socket socket = new Socket("127.0.0.1", 9100)) {
            OutputStream out = socket.getOutputStream();
            String[] cmds = {"play", "pause", "play", "stop"};
            for (String cmd : cmds) {
                byte[] msg = MediaProtocol.packCommand(cmd, null);
                out.write(msg);
                System.out.println("Sent command: " + cmd);
                Thread.sleep(1000);
            }
        }
    }
} 

File: py/media/controller.py
import socket
import time
from media_protocol import MediaControlProtocol

HOST = '127.0.0.1'
PORT = 9100

if __name__ == '__main__':
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.connect((HOST, PORT))
        cmds = ['play', 'pause', 'play', 'stop']
        for cmd in cmds:
            msg = MediaControlProtocol.pack_command(cmd)
            s.sendall(msg)
            print(f'Sent command: {cmd}')
            time.sleep(1) 